### PR TITLE
Compiler&Bundler improvement

### PR DESCRIPTION
**Pull request by ije**
- [x] **server**: support custom server 
- [x] **plugin**: rewrite plugin system, add `ServerPlugin` type to enhance server runtime
- [x] **compiler**: add source map in dev mode
- [x] **bundler**: bundle styles
- [ ] **bundler**: bundle dynamic dependencies
- [x] **bundler**: _tree-shake_ `useDeno` hook callback code in production mode
- [x] **framework/core**: improve page async deps loading
- [x] **framework/react**: add `revalidate` option for `useDeno` hook
- [x] **breaking**: remove builtin less loader (should use plugin instead)
- [x] **breaking**: remove builtin markdown pages
- [x] **breaking**: remove `SEO` and `Viewport` components

### CODE REVIEW

**Comment by shadowtime2000** [comment]
How do you play on bundling styles? writing them to a `.css` file?

**Comment by ije** [comment]
@shadowtime2000 will bundle css in js to reduce connections

**Review by shadowtime2000 — COMMENTED** [review]
[denoland/deno#9212](https://github.com/denoland/deno/issues/9212) has been closed I think you can remove the workaround

**Comment by ije** [comment]
@shadowtime2000 will remove it after deno 1.7.1 released.

**Comment by shadowtime2000** [comment]
@ije Maybe I am not understanding the architecture of the compiler/bundler, but that issue got closed after Deno updated there crates and since we are using SWC directly I believe wouldn't we just have to update SWC?

**Comment by ije** [comment]
@shadowtime2000 now we use swc to compile source code (ts/tsx/alef/vue/...), then use `deno bundle` to bundle the transpiled code.

**Comment by ije** [comment]
the compiler will transpile the source code to es2020 as esm that can be imported by browser and deno(ssr) in dev mode, then do bundling in production mode.

**Comment by ije** [comment]
custom server: 

```ts
import { serve } from "https://deno.land/std/http/servermts"
import Application from "https://deno.land/x/aleph/server/app.ts"
import { Server } from "https://deno.land/x/aleph/server/mod.ts"

const app = new Appliaction(options) 
const server = new Server(app)
const s = serve({ hostname, port })

for await (const r of s) {
  server.handle(r)
}
```

**Review by shadowtime2000 — COMMENTED** [review]
Deno 1.7.1 has been released I think you can remove the compat fixer now or else it will break. You could also alternatively have it replace with a regex that makes sure there isn't a `var` before that statement making it compatable with both versions.

File: plugins/sass_test.ts [hunk 1]
```diff
@@ -2,24 +2,24 @@ import { assertEquals } from 'https://deno.land/std@0.83.0/testing/asserts.ts'
...
 
 Deno.test('project sass loader plugin', async () => {
-    let ret = await plugin.transform(
+    let ret = await plugin.precompile!(
         (new TextEncoder).encode('$someVar: 123px\n.some-selector\n  width: 123px'),
         'test.sass'
     )
     assertEquals(ret.code, '.some-selector {\n  width: 123px;\n}')
-    ret = await plugin({ indentType: 'tab', indentWidth: 2 }).transform(
+    ret = await plugin({ indentType: 'tab', indentWidth: 2 }).precompile!(
```
Comment (shadowtime2000) [inline plugins/sass_test.ts]: Why the `!` after `precompile`? is it a typo? I don't think that is valid syntax

Comment (ije) [reply plugins/sass_test.ts]: now loader plugin's precompile method is optional, but sass one is not

**Pull request merged**